VERSIONING.md

```diff
@@ -0,0 +1,221 @@
+# Versioning
+
+This document covers Vector's versioning and what it means as a user of Vector.
+
+**Please note, Vector is currently in its pre-1.0 phase and quickly approaching
+1.0. Minor version increments can introduce breaking changes during this phase.
+Please see the [what to expect](#what-to-expect) section for more info.**
+
+<!-- MarkdownTOC autolink="true" style="ordered" indent="   " -->
+
+1. [Convention](#convention)
+1. [Public API](#public-api)
+   1. [Areas that *are* covered](#areas-that-are-covered)
+      1. [Intended for *public* consumption](#intended-for-public-consumption)
+      1. [Intended for *private* consumption](#intended-for-private-consumption)
+   1. [Areas that are *NOT* covered](#areas-that-are-not-covered)
+1. [Releases](#releases)
+   1. [Release channels](#release-channels)
+      1. [Stable channel](#stable-channel)
+      1. [Nightly channel](#nightly-channel)
+   1. [Release tracking](#release-tracking)
+      1. [Stable channel](#stable-channel-1)
+      1. [Nightly channel](#nightly-channel-1)
+   1. [Release downloading](#release-downloading)
+   1. [Release cadence](#release-cadence)
+      1. [Stable channel](#stable-channel-2)
+      1. [Nightly channel](#nightly-channel-2)
```

Comment:
Seeing this in the TOC now, I am wondering if it belongs in this document. I wouldn't have thought to look at `VERSIONING.md` to figure out the release cadence. My vote would be to still have a separate `RELEASING.md` file that contains these items as well as `Which release type should I be using?` and `Release cadence` and links to this doc for versioning info.

Reply:
Sure, we can split it out, but I think that's for the public, not internal development. So I'll add a new doc.

Reply:
Done in https://github.com/timberio/vector/pull/8795/commits/efe2048f2117b44033ee61faa761e2ed0c696620